src/main/groovy/com/github/dispader/manifesto/Version.groovy

```diff
@@ -1,18 +1,47 @@
 package com.github.dispader.manifesto
 
 class Version {
+    static String MSG_NO_REPO = "The Manifesto plugin is only useful for projects using git source control."
+    static String MSG_NO_COMMITS = "The Manifesto plugin couldn't find any git commits."
+    static String MSG_NO_TAGS = "The Manifesto plugin couldn't find any version tags. You can create your first tag with `git tag -a v0.1.0`."
 
-    private static getGit() { org.ajoberstar.grgit.Grgit.open() }
+    private static getGit() {
+        try {
+            org.ajoberstar.grgit.Grgit.open()
+        } catch (IllegalArgumentException) {
+            null
+        }
+    }
 
-    static getVersioned() { ( Version.version == '0.0.0' ) ? false : true  }
+    static getVersioned() {
+       ( Version.hasRepo && Version.hasCommits && Version.hasTags )
+    }
 
     static getVersion() {
-        String description = Version.git?.describe { }
-        if ( !description ) { return '0.0.0' }
+        if ( !Version.versioned ) { return null }
```

Comment:
Here's maybe the only criticism for the Pull— `null` is kinda evil as a public response.  Maybe throw `IllegalStateException` if returning a version makes no sense?  Or still `0.0.0` (or `unversioned`, which is a nonsensical value which we'd never want to actually use, but would be an expected?

This might just be my inner "Law of Demeter" voice, but it feels like we should respond with some version-like ugly description or just tell the folks "you can't have a response."